scripts/reindex_articles.coffee

```diff
@@ -15,15 +16,17 @@ switch process.env.NODE_ENV
 # Connect to database
 db = mongojs(process.env.MONGOHQ_URL, ['articles'])
 
-db.articles.find({ }).toArray (err, articles) ->
+db.articles.find({}).toArray (err, articles) ->
   console.log(err) if err
   indexWorker(articles, 0)
 
 indexWorker = (articles, i) ->
   a = articles[i]
   console.log('indexing ' + a._id)
-  indexForSearch Article.present(a)
-  setTimeout( =>
-    console.log('indexed ' + a.id or a._id)
-    indexWorker(articles, ++i)
-  , 30)
+  Article.present(a).then((results) =>
+    indexForSearch results
+    setTimeout( =>
+      console.log('indexed ' + a.id or a._id)
+      indexWorker(articles, ++i)
+    , 30)
+  )
```

Comment:
This file doesn't make sense to me since it's calling `indexWorker(articles, 0)` and never iterating on the array itself but I added support for the presentation layer anyway... thoughts @cavvia ?

Reply:
I use recursion here with a sleep of 30ms between each call, in order to not overwhelm ES with indexing calls. We pass the index along with the array to the timeout handler `indexWorker` and it selects the next article. There's probably a more intuitive way to do this in js but I don't think there's a native `sleep` function, which is why i use timeouts. This will error when it's done rather than exit gracefully, so it could certainly be improved!

Reply:
Ahh I see, my bad for not seeing that incrementor. Thanks @cavvia. We can use something like [`async.mapSeries`](https://caolan.github.io/async/docs.html#mapSeries) and run buckets of [parallel](https://caolan.github.io/async/docs.html#parallel) requests depending on how often we'd like to hit `indexForSearch`. I'll pair with you on it sometime if you'd like! For now this should continue to work as usual.